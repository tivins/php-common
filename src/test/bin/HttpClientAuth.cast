{"version":2,"width":80,"height":10,"timestamp":1674502454,"env":{"SHELL":"\/bin\/bash","TERM":"xterm-256color"}}
[0,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ run.php\r\n"]
[0.3951749801635742,"o","Tivins\\Core\\Net\\Client#21 200 (613)\r\n{\r\n  \"args\": {}, \r\n  \"data\": \"{\\\"yo\\\":\\\"lo\\\"}\", \r\n  \"files\": {}, \r\n  \"form\": {}, \r\n  \"headers\": {\r\n    \"Accept\": \"*\/*\", \r\n    \"Accept-Encoding\": \"deflate, gzip, br\", \r\n    \"Authorization\": \"Bearer a-token-from-elsewhere\", \r\n    \"Content-Length\": \"11\", \r\n    \"Content-Type\": \"application\/json\", \r\n    \"Host\": \"httpbin.org\", \r\n    \"User-Agent\": \"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:101.0) Gecko\/20100101 Firefox\/101.0\", \r\n    \"X-Amzn-Trace-Id\": \"Root=1-63cee136-1ec8d0b936b8d70030aa3d1a\"\r\n  }, \r\n  \"json\": {\r\n    \"yo\": \"lo\"\r\n  }, \r\n  \"method\": \"POST\", \r\n  \"origin\": \"xx.xx.xx.xx\", \r\n  \"url\": \"https:\/\/httpbin.org\/anything\"\r\n}\r\n\r\n"]
[0.39914798736572266,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ "]