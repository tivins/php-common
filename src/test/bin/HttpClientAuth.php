<?php

use Tivins\Core\Net\Client;
use Tivins\Core\Net\ClientException;
use Tivins\Core\Net\Http\Header;
use Tivins\Core\Net\Http\Headers;

require 'vendor/autoload.php';

$token = 'a-token-from-elsewhere';

$headers = (new Headers())
    ->setHeader(Header::Authorization, 'Bearer ' . $token);

try {
    $client = (new Client('https://httpbin.org/anything'))
        ->setHeaders($headers)
        ->postJSON(['yo' => 'lo'])
        ->execute();
}
catch (ClientException $e) {
    exit($e->client . ' ' . $e->getMessage() . "\n");
}

echo $client . ' ' . $client->getCode() . ' (' . strlen($client->getContent()) . ')', PHP_EOL;
echo $client->getContent(), PHP_EOL;
