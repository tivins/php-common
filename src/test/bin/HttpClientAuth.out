Tivins\Core\Net\Client#21 200 (613)
{
  "args": {}, 
  "data": "{\"yo\":\"lo\"}", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "deflate, gzip, br", 
    "Authorization": "Bearer a-token-from-elsewhere", 
    "Content-Length": "11", 
    "Content-Type": "application/json", 
    "Host": "httpbin.org", 
    "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:101.0) Gecko/20100101 Firefox/101.0", 
    "X-Amzn-Trace-Id": "Root=1-63cee136-1ec8d0b936b8d70030aa3d1a"
  }, 
  "json": {
    "yo": "lo"
  }, 
  "method": "POST", 
  "origin": "xx.xx.xx.xx", 
  "url": "https://httpbin.org/anything"
}

